`[Launchpad Discovery]`: New community filters and sorting options

mzparacha opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Link to Issue
Closes: https://github.com/hicommonwealth/commonwealth/issues/8801

## Description of Changes
- Added support for new filters on the communities explore page.
- New filters changes include
    - Added `Sort by` => possible values => `Most Recent`, `Market Cap`, `Price`, `Number of Members`, `Number of Threads`. ATM `Market Cap` and `Price` filters won't work. They will be done in https://github.com/hicommonwealth/commonwealth/issues/9694
    - Added `Sort order` => possible values => `Ascending`, `Descending`
    - Added `Community type` => possible values => `Launchpad`, `Basic`
        - when launchpad is selected it will only show communities associated to a token
        - when basic is selected it will only show communities that are not associated to any token
        - when this filter is removed it will show all communities
   - Updated `Community Ecosystem` => 3 new filters additions => `Base`, `Blast`, `Polygon` - when any of these are selected it will filter communities by the correct eth/cosmos chain id
   - Updated `Community Ecosystem` => 2 filter removals => `Substrate` and `NEAR`
- Some filter UI elements / updated labels are not behind the feature flag, this is intentional.

## "How We Fixed It"
N/A

## Test Plan
1. Enable `FLAG_TOKENIZED_COMMUNITY`
2. Visit the `/communities` page
3. Verify you see updated layout for filters section (I didn't add this behind feature flag intentionally, this is some UI only change that doesn't introduce new features)
4. Verify you are able to see new filter changes in the filters drawer - these should only be visible when `FLAG_TOKENIZED_COMMUNITY` is enabled
5. Verify that each filter you apply works as described in the description section above.
6. Verify that you can combine multiple filters to narrow down results ex: `sort "Basic" community type on "Base" ecosystem, that have self identified with "DAO" preference and "Order by" Number of members with most members first (Order direction = descending)`

## Deployment Plan
N/A

## Other Considerations
N/A

Rotorsoft reviewed (approved):
BE looks good

masvelio reviewed (approved):
1. Getting error on `http://localhost:8080/api/profile/v2` - probably you need latest master
2. Mobile is still not usable when `Launch an idea ` banner is visible
3. should sort order have some default state? Or ant other filter should be selected by default, like sort by? 
4. When I click only ascending or descending, getting 500. Actually, what parameter we take into consideration if we only enable this filter?
5. When I pick some filter eg `community type` and 'ascending' and then I reset it by clicking X on the button, `community type` is reset but order is still 'ascending'. Somehow related to numer 3 above. 
6. When I hover over X, cursor changes to pointer but I can click anywhere in the button and the filter gets deleted. We should either make pointer on the whole button or make it clickable only on X.
![image](https://github.com/user-attachments/assets/191867a1-2245-4593-aeb1-ead9a7af531d)

Most of these things above can be found on this video 

https://github.com/user-attachments/assets/0b224818-5a13-4e11-b4f8-0260cc9fbc05



Codewise looks good. Some small comments, I will approve in advance to unblock but lets try to resolve them.

mzparacha commented:
Re: https://github.com/hicommonwealth/commonwealth/pull/9738#pullrequestreview-2410323744

1. Master merged, should be fixed now.
2. I understand the many vertical spacing issues on the communities page, we will have many more in upcoming PRs, until the design gets updated to properly accommodate all the display elements on this page. I will come back to this after completing the UI stuff
3. By default the API sorts by `DESC` order, but we don't show it in frontend, I say its not a big deal but lmk if otherwise
4. This should set the `profile_count` as the order in API if nothing is provided in frontend.
5. Updated, now it should remove sort order filter, when sort by filter is removed.
6. Fixed now.

Thanks for identifying these!

cc: @masvelio
